code/game/objects/items/implants/implant_freedom.dm

```diff
@@ -5,35 +5,44 @@
 	implant_color = "r"
 	uses = 4
 
+/obj/item/implant/freedom/implant(mob/living/target, mob/user, silent, force)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!iscarbon(target)) //This is pretty much useless for anyone else since they can't be cuffed
+		balloon_alert(user, "that would be a waste!")
+		return FALSE
+	return TRUE
 
 /obj/item/implant/freedom/activate()
 	. = ..()
+	var/mob/living/carbon/carbon_imp_in = imp_in
+	if(!carbon_imp_in.handcuffed && !carbon_imp_in.legcuffed)
+		balloon_alert(carbon_imp_in, "no restraints!")
+		return
+
 	uses--
-	to_chat(imp_in, span_hear("You feel a faint click."))
-	if(iscarbon(imp_in))
-		var/mob/living/carbon/C_imp_in = imp_in
-		C_imp_in.uncuff()
+
+	balloon_alert(carbon_imp_in, "bindings released!")
```

Comment:
```suggestion
```
- Don't need a balloon alert for success, there's already visual feedback here.
- `!` is reserved for negative/failure notification.